bindings/mqtt/mqtt.go

```diff
@@ -243,58 +243,96 @@ func (m *MQTT) handleMessage(ctx context.Context, handler bindings.Handler, mqtt
 }
 
 func (m *MQTT) Read(ctx context.Context, handler bindings.Handler) error {
-	// reset synchronization
-	if m.consumer != nil {
-		m.logger.Warnf("re-initializing the subscriber")
-		m.consumer.Disconnect(5)
-		m.consumer = nil
+	// mqtt broker allows only one connection at a given time from a clientID
+	topicHash := sha256.Sum256([]byte(m.metadata.topic))
+	consumerClientID := fmt.Sprintf("%s-consumer-%x", m.metadata.clientID, topicHash[0:4])
```

Comment:
This is a big change in how consumerID are calculated. Why is this needed?

Reply:
There can only be 1 active consumer with the same name. If you subscribe to multiple topics with the same binding, then it would fail.

Note that this change impact the binding only, which is not a stable component. If that's a concern for a hotfix, I can move that into a separate PR and have the pubsub only in a hotfix